main/src/rootls.cxx

```diff
@@ -0,0 +1,743 @@
+/// \file rootls.cxx
+///
+/// Native implementation of rootls, partially based on rootls.py.
+///
+/// \author Giacomo Parolini <giacomo.parolini@cern.ch>
+/// \date 2025-06-27
+
+/*************************************************************************
+ * Copyright (C) 1995-2020, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <algorithm>
+#include <cstdint>
+#include <cstring>
+#include <deque>
+#include <iomanip>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "wildcards.hpp"
+
+#include <TBranch.h>
+#include <TError.h>
+#include <TFile.h>
+#include <TKey.h>
+#include <TTree.h>
+
+#include <ROOT/StringUtils.hxx>
+#include <ROOT/RError.hxx>
+
+#if defined(R__UNIX)
+#include <sys/ioctl.h>
+#include <unistd.h>
+#elif defined(R__WIN32)
+#define WIN32_LEAN_AND_MEAN
+#define VC_EXTRALEAN
+#include <windows.h>
+#endif
+
+static const char *const kAnsiNone = "\x1B[0m";
```

Comment:
subjective: I would prefer `constexpr auto`.

Reply:
I'm never quite sure what `auto` becomes with string literals (I assume `const char *`) so I prefer being explicit in these cases